plotly_resampler/figure_resampler/figurewidget_resampler.py

```diff
@@ -259,22 +259,23 @@ def _update_spike_ranges(self, layout, *showspikes, force_update=False):
                 self._relayout_hist.append(["showspikes-update", len(update_data) - 1])
                 self._relayout_hist.append("-" * 30)
 
-            with self.batch_update():
-                # First update the layout (first item of update_data)
-                if not force_update:
-                    self.layout.update(self._parse_relayout(update_data[0]))
-
-                # Also:  Remove the showspikes from the layout, otherwise the autorange
-                # will not work as intended (it will not be triggered again)
-                # Note: this removal causes a second trigger of this method
-                # which will go in the "else" part below.
-                for xaxis_str in self._xaxis_list:
-                    self.layout[xaxis_str].pop("showspikes")
+            if not self._is_no_update(update_data):
+                with self.batch_update():
+                    # First update the layout (first item of update_data)
+                    if not force_update:
+                        self.layout.update(self._parse_relayout(update_data[0]))
+
+                    # Also: Remove the showspikes from the layout, otherwise the autorange
+                    # will not work as intended (it will not be triggered again)
+                    # Note: this removal causes a second trigger of this method
+                    # which will go in the "else" part below.
+                    for xaxis_str in self._xaxis_list:
+                        self.layout[xaxis_str].pop("showspikes")
```

Comment:
Not 100% sure whether this is necessary when `update_data` is NoUpdate
(In the previous logic, this would be executed)

Reply:
Quite sure it is correct to only execute this code if update_data is a proper update.